operator-pipeline-images/operatorcert/pyxis.py

```diff
@@ -289,3 +290,82 @@ def get_repository_by_isv_pid(base_url: str, isv_pid: str) -> Any:
         raise
     json_resp = resp.json()
     return None if len(json_resp.get("data")) == 0 else json_resp["data"][0]
+
+
+def post_image_request(
+    base_url: str,
+    project_id: str,
+    image_id: str,
+    operation: str,
+) -> Any:
+    """
+    POST image request to Pyxis
+
+    Args:
+        base_url (str): base Pyxis URL,
+        project_id (str): Certification project _id
+        image_id (str): Container image _id
+        operation (str): Operation type
+
+    Returns:
+        Any: Image request response
+    """
+
+    payload = {
+        "operation": operation,
+        "image_id": image_id,
+    }
+
+    image_request_url = urljoin(
+        base_url, f"/v1/projects/certification/id/{project_id}/requests/images"
+    )
+
+    image_request = post(image_request_url, payload)
+
+    return image_request
+
+
+def wait_for_image_request(base_url: str, image_request_id: str) -> Any:
+    """
+    Wait for image request to finish and transition to one of the final states.
+    The request is tracked and status is printed every 5 seconds.
```

Comment:
Shouldn't we also have a deadline for this function to terminate? If, for some reason, the image request never transitions to one of the expected final statuses, we keep hammering pyxis...

Reply:
I added a timeout and configurable interval.